Components/FlightControl/WatchdogTask.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ ******************************************************************************
+ * File Name          : WatchdogTask.cpp
+ * Description        : Primary flight task, default task for the system.
+ ******************************************************************************
+*/
+#include "GPIO.hpp"
+#include "SystemDefines.hpp"
+#include "Timer.hpp"
+#include "WatchdogTask.hpp"
+#include "FlightTask.hpp"
+
+
+/**
+ * @brief Initialize the WatchdogTask
+ * @params Must pass in the timer period, If no heartbeat in received within this period then the timerr will be reset
+ */
+WatchdogTask::WatchdogTask(){}
+
+/**
+ * @brief Initialize the FlightTask
+ */
+void WatchdogTask::InitTask()
+{
+    // Make sure the task is not already initialized
+    SOAR_ASSERT(rtTaskHandle == nullptr, "Cannot initialize flight task twice");
+
+    BaseType_t rtValue =
+        xTaskCreate((TaskFunction_t)WatchdogTask::RunTask,
+            (const char*)"WatchdogTask",
+            (uint16_t)WATCHDOG_TASK_STACK_DEPTH_WORDS,
+            (void*)this,
+            (UBaseType_t)WATCHDOG_TASK_RTOS_PRIORITY,
+            (TaskHandle_t*)&rtTaskHandle);
+
+    SOAR_ASSERT(rtValue == pdPASS, "WatchdogTask::InitTask() - xTaskCreate() failed");
+}
+
+/**
+ * @brief This function is called if the heartbeat timer expires and it sends a command to reset the system
+ * @param Automatically passes is timer handle for callback, should not be used
+ */
+void WatchdogTask::HeartbeatFailureCallback(TimerHandle_t rtTimerHandle)
+{
+
+    SOAR_PRINT("The system lost its heartbeat and had to reset!!!\n");
+//    SOAR_ASSERT(false ,"The system lost its heartbeat and had to reset!!! \n");
+    Timer::DefaultCallback(rtTimerHandle);
+    WatchdogTask::Inst().SendCommand(Command(CONTROL_ACTION, RSC_ANY_TO_ABORT));
+}
+
+/**
+ * @brief Handles a command
+ * @param cm Command reference to handle
+ */
+void WatchdogTask::HandleCommand(Command& cm)
+{
+    switch (cm.GetCommand()) {
+    case REQUEST_COMMAND: {
```

Comment:
I'd probably make a new entry in the GLOBAL_COMMAND enum called HEARTBEAT_COMMAND so it's a bit more clear

Reply:
i think i did it